File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let n = heights.length - 1 

        let i = 0
        let maxArea = 0
        let j = n

        while(i<j){
            let minH = Math.min(heights[i], heights[j])

            let width = j - i

            let area = minH * width

            maxArea = Math.max(maxArea, area)

            if(heights[i] < heights[j] ){
                i++
            }
            else{
                j--
            }
        }
        return maxArea 
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        maxWater = 0
        while left < right:

            minH = min (heights[left], heights[right])

            contain = (right - left) * minH

            if heights[left] < heights[right]:
                left +=1

            else:
                right -=1

            maxWater = max(maxWater, contain)
        return maxWater
        